packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -0,0 +1,443 @@
+
+    fireEvent.click(nextButton!);
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+
+    fireEvent.click(nextButton!);
+    fireEvent.click(nextButton!);
+
+    expect(onChange).toHaveBeenLastCalledWith(3);
+
+    expect(queryByRole('button', { name: 'Next Slide' })).not.toBeInTheDocument();
+    expect(queryByRole('button', { name: 'Previous Slide' })).toBeInTheDocument();
+  });
+
+  test('when visibleItems:autofit & shouldAddStartEndSpacing is undefined then we hide the indicators since they are useless', () => {
+    const onChange = jest.fn();
+    const { queryAllByRole } = renderWithTheme(
+      <Carousel onChange={onChange} visibleItems="autofit">
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(queryAllByRole('tab').length).toBe(0);
+  });
+});
+
+describe('Carousel Snapshots', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(
+      <Carousel>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render with visibleItems', () => {
+    const { container } = renderWithTheme(
+      <Carousel visibleItems={2}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render with shouldAddStartEndSpacing', () => {
+    const { container } = renderWithTheme(
+      <Carousel carouselItemWidth={{ base: '90%', m: '300px' }} shouldAddStartEndSpacing>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render with showOverlay', () => {
+    const { container } = renderWithTheme(
+      <Carousel scrollOverlayColor="surface.background.level1.lowContrast" shouldAddStartEndSpacing>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
```

Comment:
add test case for not rendering overlay when on mobile?

Reply:
done.